pkg/utils/mainhelper/init.go

```diff
@@ -92,9 +96,25 @@ func ProcessEnvConfigOrDie(env interface{}) {
 	log.Printf("Running with env: %+v", env)
 }
 
+// ProcessGCPProfilerEnvConfigOrDie tries to enable the GCP Profiler if env vars
+// have configured it.
+func ProcessGCPProfilerEnvConfigOrDie(component string, logger *zap.SugaredLogger) {
+	var env profiling.GCPProfilerEnvConfig
+	if err := envconfig.Process("", &env); err != nil {
+		logger.Desugar().Fatal("Failed to process GCP Profiler env config", zap.Error(err))
+	}
+	started, err := profiling.StartGCPProfiler(component, env)
+	if err != nil {
+		logger.Desugar().Fatal("Failed to start GCP Profiler", zap.Error(err))
+	}
+	if started {
```

Comment:
nit: It seems that the `started` is only used for logging. It can be removed if you check `GCPProfilerEnabled` and call `StartGCPProfiler` if true.

Reply:
I switched to that style and added a method to check enablement.